src/parsers/manifest/utils/get_maximum_position.ts

```diff
@@ -15,18 +15,40 @@
  */
 
 import log from "../../../log";
-import { IParsedManifest } from "../types";
+import {
+  IParsedPartialPeriod,
+  IParsedPeriod,
+} from "../types";
 import getLastPositionFromAdaptation from "./get_last_time_from_adaptation";
 
+export interface INeedsToLoadPeriodEvent {
+  type : "needs-to-load-period";
+  value : IParsedPartialPeriod;
+}
+
+export interface IResultEvent {
+  type : "result";
+  value : number | undefined;
+}
+
 /**
- * @param {Object} manifest
- * @returns {number | undefined}
+ * @param {Array.<Object>} periods
+ * @returns {Object}
  */
+export default function getMaximumPosition(periods : IParsedPeriod[]) : IResultEvent;
 export default function getMaximumPosition(
-  manifest: IParsedManifest
-) : number | undefined {
-  for (let i = manifest.periods.length - 1; i >= 0; i--) {
-    const periodAdaptations = manifest.periods[i].adaptations;
+  periods : Array<IParsedPartialPeriod | IParsedPeriod>
+) : INeedsToLoadPeriodEvent | IResultEvent;
+export default function getMaximumPosition(
+  periods : Array<IParsedPartialPeriod | IParsedPeriod>
+) : INeedsToLoadPeriodEvent | IResultEvent {
+  for (let i = periods.length - 1; i >= 0; i--) {
+    const period = periods[i];
+    if (!period.isLoaded) {
```

Comment:
Why not just testing last period ?

Reply:
The role of this loop is to continue with the "before-last" period if the last was found to be in the future etc.

(It's in the future - or empty - because its `getLastPosition` call returns `null`)

Reply:
So here only needed periods are testes. We will in any case return as soon as we find a credible last position